libfreerdp/core/multitransport.c

```diff
@@ -17,35 +17,155 @@
  * limitations under the License.
  */
 
+#include <winpr/assert.h>
 #include <freerdp/config.h>
+#include <freerdp/log.h>
 
+#include "rdp.h"
 #include "multitransport.h"
 
-#include <freerdp/log.h>
-
 #define TAG FREERDP_TAG("core.multitransport")
 
-int rdp_recv_multitransport_packet(rdpRdp* rdp, wStream* s)
+struct rdp_multitransport
+{
+	rdpRdp* rdp;
+
+	UINT32 requestId;         /* requestId (4 bytes) */
+	UINT16 requestedProtocol; /* requestedProtocol (2 bytes) */
+	UINT16 reserved;          /* reserved (2 bytes) */
+	BYTE securityCookie[16];  /* securityCookie (16 bytes) */
+};
+
+static BOOL compare(const rdpMultitransport* srv, const rdpMultitransport* client)
```

Comment:
Perhaps we can have a less generic name than `compare` ...

Reply:
`true`, did not think about that as the function is `static`